core/package/__tests__/core-package.test.js

```diff
@@ -273,6 +274,37 @@ describe("Package", () => {
       );
     });
   });
+
+  describe(".updateLocalDependency()", () => {
+    it("works with workspace: protocols", () => {
+      const pkg = factory({
+        dependencies: {
+          a: "workspace:^1.0.0",
+          b: "workspace:^1.0.0",
+          c: "workspace:./foo",
+          d: "file:./foo",
+          e: "^1.0.0",
+        },
+      });
+
+      const resolved = npa.resolve("a", "^1.0.0", ".");
+      resolved.explicitWorkspace = true;
+
+      pkg.updateLocalDependency(resolved, "2.0.0", "^");
+
+      expect(pkg.toJSON()).toMatchInlineSnapshot(`
+        Object {
+          "dependencies": Object {
+            "a": "workspace:^2.0.0",
```

Comment:
One wrinkle, however, is that _sometimes_ `pkg.updateLocalDependency()` is intended to overwrite any protocol (such as `file:`) with the resolved version. Existing code has cheesy conditionals that exclude the call when the `resolved.type === "directory"`, so we're gonna have to make this method smarter somehow (or split the intent into two methods...)